pkg/mermaid/mermaidsequencediagram.go

```diff
@@ -0,0 +1,157 @@
+package mermaid
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/anz-bank/sysl/pkg/sysl"
+)
+
+type sequencepair struct {
+	firstapp, secondapp, endpoint string
+}
+
+const projDir = "../../"
+
+var startElse = regexp.MustCompile("^else.*")
+var isElse = regexp.MustCompile("^else$")
+var isLoop = regexp.MustCompile("^loop.*")
+
+func GenerateMermaidSequenceDiagram(m *sysl.Module, appname string, epname string) (string, error) {
+	var sp []sequencepair
+	return generateMermaidSequenceDiagramHelper(m, appname, epname, "...", 1, &sp, true)
+}
+
+func generateMermaidSequenceDiagramHelper(m *sysl.Module, appname string, epname string,
+	previousapp string, indent int, sequencepairs *[]sequencepair, thestart bool) (string, error) {
+	var result string
+	if thestart {
+		result = "%% AUTOGENERATED CODE -- DO NOT EDIT!\n\nsequenceDiagram\n"
+		if err := validAppNameAndEndpoint(m, appname, epname); err != nil {
+			return "", err
+		}
+		result += fmt.Sprintf(" %s->>%s: %s\n", previousapp, appname, epname)
+	}
+	statements := m.Apps[appname].Endpoints[epname].GetStmt()
+	result += printSequenceDiagramStatements(m, statements, appname, previousapp, indent, sequencepairs, thestart)
+	return result, nil
+}
+
+func printSequenceDiagramStatements(m *sysl.Module, statements []*sysl.Statement, appname string,
```

Comment:
This seems to be a pretty meaty function; do you think you could split off some of the switch cases of into their own functions?

Reply:
I did add a new function for a few but it seemed like it wasn't taking away much so I preferred to leave everything together.